### PR TITLE
Validation fixes for Moments and Strategies

### PR DESCRIPTION
**Pull request by thomasoboyle**
### Change Log
- Move validations from Moments & Strategies forms into their own partials. As stated in https://github.com/julianguyen/ifme/pull/716,
moving the validations into partial triggers validation messages prior to page reload. Validations now work properly for both create and edit for Moments & Strategies.
- Add myself as a contributor ;)

### CODE REVIEW

File: app/views/moments/_form.html.erb [hunk 1]
```diff
@@ -1,11 +1,6 @@
 <%= render partial: 'shared/editor_modal' %>
 
 <%= form_for(@moment) do |f| %>
-<% if @moment.errors.any? %>
```
Comment (julianguyen) [inline app/views/moments/_form.html.erb]: Similar comment made in the other PR, can't we put `<%= render 'validation_errors' %>` here?

File: app/views/strategies/_form.html.erb [hunk 2]
```diff
@@ -1,11 +1,6 @@
 <%= render partial: 'shared/editor_modal' %>
 
 <%= form_for(@strategy) do |f| %>
-<% if @strategy.errors.any? %>
```
Comment (julianguyen) [inline app/views/strategies/_form.html.erb]: Put `<%= render 'validation_errors' %>` here instead?

**Comment by julianguyen** [comment]
You should also rebase from master! :)

**Comment by thomasoboyle** [comment]
Yay! Thank you!

**Pull request merged**